# Light Theme and Dark Theme: Case Study

In [2]:
import pandas as pd
import matplotlib.pyplot as plt # matplotlib.pyplot is a plotting library in Python that produces publication-quality 
import seaborn as sns # seaborn is a Python data visualization library based on matplotlib.
from scipy.stats import ttest_ind
import os

In [4]:
df=pd.read_csv(r'D:\Project\New folder\website_ab_test.csv')

In [5]:
print(df)

           Theme  Click Through Rate  Conversion Rate  Bounce Rate  \
0    Light Theme            0.054920         0.282367     0.405085   
1    Light Theme            0.113932         0.032973     0.732759   
2     Dark Theme            0.323352         0.178763     0.296543   
3    Light Theme            0.485836         0.325225     0.245001   
4    Light Theme            0.034783         0.196766     0.765100   
..           ...                 ...              ...          ...   
995   Dark Theme            0.282792         0.401605     0.200720   
996   Dark Theme            0.299917         0.026372     0.762641   
997  Light Theme            0.370254         0.019838     0.607136   
998  Light Theme            0.095815         0.137953     0.458898   
999   Dark Theme            0.342588         0.061315     0.452410   

     Scroll_Depth  Age   Location  Session_Duration Purchases Added_to_Cart  
0       72.489458   25    Chennai              1535        No           Yes  
1  

In [6]:
df.head()

Theme  Click Through Rate  Conversion Rate  Bounce Rate  \
0  Light Theme            0.054920         0.282367     0.405085   
1  Light Theme            0.113932         0.032973     0.732759   
2   Dark Theme            0.323352         0.178763     0.296543   
3  Light Theme            0.485836         0.325225     0.245001   
4  Light Theme            0.034783         0.196766     0.765100   

   Scroll_Depth  Age   Location  Session_Duration Purchases Added_to_Cart  
0     72.489458   25    Chennai              1535        No           Yes  
1     61.858568   19       Pune               303        No           Yes  
2     45.737376   47    Chennai               563       Yes           Yes  
3     76.305298   58       Pune               385       Yes            No  
4     48.927407   25  New Delhi              1437        No            No

In [7]:
df.isnull()

Theme  Click Through Rate  Conversion Rate  Bounce Rate  Scroll_Depth  \
0    False               False            False        False         False   
1    False               False            False        False         False   
2    False               False            False        False         False   
3    False               False            False        False         False   
4    False               False            False        False         False   
..     ...                 ...              ...          ...           ...   
995  False               False            False        False         False   
996  False               False            False        False         False   
997  False               False            False        False         False   
998  False               False            False        False         False   
999  False               False            False        False         False   

       Age  Location  Session_Duration  Purchases  Added_to_Cart  
0    False     False             False      False          False  
1    False     False             False      False          False  
2    False     False             False      False          False  
3    False     False             False      False          False  
4    False     False             False      False          False  
..     ...       ...               ...        ...            ...  
995  False     False             False      False          False  
996  False     False             False      False          False  
997  False     False             False      False          False  
998  False     False             False      False          False  
999  False     False             False      False          False  

[1000 rows x 10 columns]

In [8]:
df.shape

(1000, 10)

In [11]:
df.isna()

Theme  Click Through Rate  Conversion Rate  Bounce Rate  Scroll_Depth  \
0    False               False            False        False         False   
1    False               False            False        False         False   
2    False               False            False        False         False   
3    False               False            False        False         False   
4    False               False            False        False         False   
..     ...                 ...              ...          ...           ...   
995  False               False            False        False         False   
996  False               False            False        False         False   
997  False               False            False        False         False   
998  False               False            False        False         False   
999  False               False            False        False         False   

       Age  Location  Session_Duration  Purchases  Added_to_Cart  
0    False     False             False      False          False  
1    False     False             False      False          False  
2    False     False             False      False          False  
3    False     False             False      False          False  
4    False     False             False      False          False  
..     ...       ...               ...        ...            ...  
995  False     False             False      False          False  
996  False     False             False      False          False  
997  False     False             False      False          False  
998  False     False             False      False          False  
999  False     False             False      False          False  

[1000 rows x 10 columns]

In [33]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 10 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Theme               1000 non-null   object 
 1   Click Through Rate  1000 non-null   float64
 2   Conversion Rate     1000 non-null   float64
 3   Bounce Rate         1000 non-null   float64
 4   Scroll_Depth        1000 non-null   float64
 5   Age                 1000 non-null   int64  
 6   Location            1000 non-null   object 
 7   Session_Duration    1000 non-null   int64  
 8   Purchases           1000 non-null   object 
 9   Added_to_Cart       1000 non-null   object 
dtypes: float64(4), int64(2), object(4)
memory usage: 78.2+ KB


In [31]:
print((df.index))

RangeIndex(start=0, stop=1000, step=1)


In [26]:
print(list(df.columns))

['Theme', 'Click Through Rate', 'Conversion Rate', 'Bounce Rate', 'Scroll_Depth', 'Age', 'Location', 'Session_Duration', 'Purchases', 'Added_to_Cart']


In [27]:
df.columns

Index(['Theme', 'Click Through Rate', 'Conversion Rate', 'Bounce Rate',
       'Scroll_Depth', 'Age', 'Location', 'Session_Duration', 'Purchases',
       'Added_to_Cart'],
      dtype='object')

In [37]:
summary = {
    'Number of Records': df.shape[0],
    'Number of Columns': df.shape[1],
    'Missing Values': df.isnull().sum(),
    'Numerical Columns Summary': df.describe()
}

summary

{'Number of Records': 1000,
 'Number of Columns': 10,
 'Missing Values': Theme                 0
 Click Through Rate    0
 Conversion Rate       0
 Bounce Rate           0
 Scroll_Depth          0
 Age                   0
 Location              0
 Session_Duration      0
 Purchases             0
 Added_to_Cart         0
 dtype: int64,
 'Numerical Columns Summary':        Click Through Rate  Conversion Rate  Bounce Rate  Scroll_Depth  \
 count         1000.000000      1000.000000  1000.000000   1000.000000   
 mean             0.256048         0.253312     0.505758     50.319494   
 std              0.139265         0.139092     0.172195     16.895269   
 min              0.010767         0.010881     0.200720     20.011738   
 25%              0.140794         0.131564     0.353609     35.655167   
 50%              0.253715         0.252823     0.514049     51.130712   
 75%              0.370674         0.373040     0.648557     64.666258   
 max              0.499989         0.49891

The dataset contains 1,000 records across 10 columns, with no missing values. Here’s a quick summary of the numerical columns:

Click Through Rate: Ranges from about 0.01 to 0.50 with a mean of approximately 0.26.

Conversion Rate: Also ranges from about 0.01 to 0.50 with a mean close to the Click Through Rate, approximately 0.25.

Bounce Rate: Varies between 0.20 and 0.80, with a mean around 0.51.

Scroll Depth: Shows a spread from 20.01 to nearly 80, with a mean of 50.32.

Age: The age of users ranges from 18 to 65 years, with a mean age of about 41.5 years.

Session Duration: This varies widely from 38 seconds to nearly 1800 seconds (30 minutes), with a mean session duration of approximately 925 seconds (about 15 minutes).

In [38]:
theme_performance = df.groupby('Theme').mean()

In [39]:
theme_performance_sorted = theme_performance.sort_values(by='Conversion Rate', ascending=False)

In [40]:
print(theme_performance_sorted)

             Click Through Rate  Conversion Rate  Bounce Rate  Scroll_Depth  \
Theme                                                                         
Light Theme            0.247109         0.255459     0.499035     50.735232   
Dark Theme             0.264501         0.251282     0.512115     49.926404   

                   Age  Session_Duration  
Theme                                     
Light Theme  41.734568        930.833333  
Dark Theme   41.332685        919.482490  


Click Through Rate (CTR): The Dark Theme has a slightly higher average CTR (0.2645) compared to the Light Theme (0.2471).
Conversion Rate: The Light Theme leads with a marginally higher average Conversion Rate (0.2555) compared to the Dark Theme (0.2513).
Bounce Rate: The Bounce Rate is slightly higher for the Dark Theme (0.5121) than for the Light Theme (0.4990).
Scroll Depth: Users on the Light Theme scroll slightly further on average (50.74%) compared to those on the Dark Theme (49.93%).
Age: The average age of users is similar across themes, with the Light Theme at approximately 41.73 years and the Dark Theme at 41.33 years.
Session Duration: The average session duration is slightly longer for users on the Light Theme (930.83 seconds) than for those on the Dark Theme (919.48 seconds).

# Getting Started with Hypothesis Testing

In [42]:
conversion_rates_light = df[df['Theme'] == 'Light Theme']['Conversion Rate']
conversion_rates_dark = df[df['Theme'] == 'Dark Theme']['Conversion Rate']

t_stat, p_value = ttest_ind(conversion_rates_light, conversion_rates_dark, equal_var=False)

t_stat, p_value


(0.4748494462782632, 0.6349982678451778)

The result of the two-sample t-test gives a p-value of approximately 0.635. Since this p-value is much greater than our significance level of 0.05, we do not have enough evidence to reject the null hypothesis. Therefore, we conclude that there is no statistically significant difference in Conversion Rates between the Light Theme and Dark Theme based on the data provided.

Null Hypothesis (H0​): There is no difference in Click Through Rates between the Light Theme and Dark Theme.
    
Alternative Hypothesis (Ha​): There is a difference in Click Rates between the Light Theme and Dark Theme.

In [43]:
# extracting click through rates for both themes
ctr_light = df[df['Theme'] == 'Light Theme']['Click Through Rate']
ctr_dark = df[df['Theme'] == 'Dark Theme']['Click Through Rate']

# performing a two-sample t-test
t_stat_ctr, p_value_ctr = ttest_ind(ctr_light, ctr_dark, equal_var=False)

t_stat_ctr, p_value_ctr

(-1.9781708664172253, 0.04818435371010704)

In [44]:
# extracting bounce rates for both themes
bounce_rates_light = df[df['Theme'] == 'Light Theme']['Bounce Rate']
bounce_rates_dark = df[df['Theme'] == 'Dark Theme']['Bounce Rate']

# performing a two-sample t-test for bounce rate
t_stat_bounce, p_value_bounce = ttest_ind(bounce_rates_light, bounce_rates_dark, equal_var=False)

# extracting scroll depths for both themes
scroll_depth_light = df[df['Theme'] == 'Light Theme']['Scroll_Depth']
scroll_depth_dark = df[df['Theme'] == 'Dark Theme']['Scroll_Depth']

# performing a two-sample t-test for scroll depth
t_stat_scroll, p_value_scroll = ttest_ind(scroll_depth_light, scroll_depth_dark, equal_var=False)

# creating a table for comparison
comparison_table = pd.DataFrame({
    'Metric': ['Click Through Rate', 'Conversion Rate', 'Bounce Rate', 'Scroll Depth'],
    'T-Statistic': [t_stat_ctr, t_stat, t_stat_bounce, t_stat_scroll],
    'P-Value': [p_value_ctr, p_value, p_value_bounce, p_value_scroll]
})

comparison_table

Metric  T-Statistic   P-Value
0  Click Through Rate    -1.978171  0.048184
1     Conversion Rate     0.474849  0.634998
2         Bounce Rate    -1.201888  0.229692
3        Scroll Depth     0.756228  0.449692

Click Through Rate: The test reveals a statistically significant difference, with the Dark Theme likely performing better (P-Value = 0.048).

Conversion Rate: No statistically significant difference was found (P-Value = 0.635).

Bounce Rate: There’s no statistically significant difference in Bounce Rates between the themes (P-Value = 0.230).

Scroll Depth: Similarly, no statistically significant difference is observed in Scroll Depths (P-Value = 0.450).

In summary, while the two themes perform similarly across most metrics, the Dark Theme has a slight edge in terms of engaging users to click through. For other key performance indicators like Conversion Rate, Bounce Rate, and Scroll Depth, the choice between a Light Theme and a Dark Theme does not significantly affect user behaviour according to the data provided.